src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift

```diff
@@ -965,12 +965,18 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding, SecureSQLStore {
 					circumstances: queryResult.string(forColumn: "locationVisitCircumstances") ?? "",
 					checkinId: checkinId
 				)
+
+				// Persisting empty Data to a BLOB field leads to retrieving nil when reading it.
+				// Because of that, we map nil to empty Data. Because "traceLocationId" is defined as NOT NULL, there should never be nil stored.
+				// For more information about that problem, please see the issue opened here: https://github.com/ccgus/fmdb/issues/73
+				let traceLocationId = queryResult.data(forColumn: "traceLocationId") ?? Data()
```

Comment:
Why can't this be `NULL`?

Reply:
Thanks! Good catch. This can and should be null. I changed it.